functional/FunctionalTest.java

```diff
@@ -899,7 +921,24 @@ public static void presignedGetObject_test2() throws Exception {
         .url(HttpUrl.parse(urlString))
         .method("GET", null)
         .build();
+
     OkHttpClient transport = new OkHttpClient();
+    SSLContext context = SSLContext.getInstance("TLSv1.2");
+
+    try {
+      context.init(null, null, null);
+    } catch (KeyManagementException e) {
+      throw new InternalException("SSL Context initialization failed.");
+    }
+
+    SSLSocketFactory socketFac = context.getSocketFactory();
+    ConnectionSpec spec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+        .tlsVersions(TlsVersion.TLS_1_2)
+        .build();
+
+    transport.setConnectionSpecs(Collections.singletonList(spec));
+    transport.setSslSocketFactory(socketFac);
+
```

Comment:
This code block can be refactored into a separate method, instead of repeating for each presigned url test cases.